frontend/src/metabase/lib/formatting.js

```diff
@@ -252,6 +252,30 @@ function formatNumberCompact(value: number, options: FormattingOptions) {
   if (options.number_style === "percent") {
     return formatNumberCompactWithoutOptions(value * 100) + "%";
   }
+  if (options.number_style === "currency") {
+    try {
+      const { value: currency } = numberFormatterForOptions(options)
+        .formatToParts(value)
+        .find(p => p.type === "currency");
+
+      // this special case ensures the "~" comes before the currency
```

Comment:
It's actually not safe to assume currency comes before the number. If you set the `currency_style` to `name` it will format like `100 US Dollars` (maybe also depending on the locale)

I'm not sure but it might be safe to assume the currency part comes first or last, which would also avoid iterating over all the parts, though you'll also want to check for a `literal` whitespace character https://runkit.com/tlrobinson/5cef1a906e2c4b001c2a55b8

Reply:
Good catch! I had totally ignored this UI:
![image](https://user-images.githubusercontent.com/691495/58598998-7eeb8380-824c-11e9-88f2-4c6462184d91.png)

What do you think about always using `currencyDisplay: "symbol"` for compact formatting? It would be more compact I guess 🤷‍♂  I _think_ symbols will always come before the number, but I can do more testing.

As you point out, locale is another place where this breaks. I think that's ok because we currently have `"en"` hardcoded. Do you think we'll change that?

Reply:
Yeah, I think that seems reasonable.

Reply:
OK sweet. It looks safe with the list of currencies in our app. 

![image](https://user-images.githubusercontent.com/691495/58599401-20bfa000-824e-11e9-8c29-3f067a4d2e72.png)

I'm actually kinda surprised that nothing was like ¢ where it comes after the number.